### ISSUE TITLE
rancher cli can not ssh to nodes in EC2

### ISSUE THREAD

**Issue by h-dang** [rancher/rancher#13556]
rancher/rancher: 2.0

I tried to use rancher-cli 2.0 to ssh to nodes
error
```
rancher ssh c-mc9gp:m-pzg4n
ssh: connect to host 192.168.0.5 port 22: Operation timed out
```
ip here is internal ip of worker node.
Is this a bug? or I have to config others?

many thanks~

**Comment by geokhoury** [comment]
I am also having the same issue with version `v2.0.1-rc1`. I get the same error, and I also was not able to manually download the keys, and use the PKI to login.

**Comment by geokhoury** [comment]
@h-dang try using the `v2.0.1-rc4` the issue seems to be solved in that release candidate.

**Comment by ajfriesen** [comment]
I have this problem too.
I am logged in the rancher cli.

```
rancher ssh NODE_ID:NODE_NAME
OpenSSH_7.2p2 Ubuntu-4ubuntu2.4, OpenSSL 1.0.2g  1 Mar 2016
debug1: Reading configuration data /home/vagrant/.ssh/config
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: Applying options for *
debug1: Connecting to 172.27.5.87 [172.27.5.87] port 22.
```

The node I try to get into is in a private subnet in aws.

**Comment by loganhz** [comment]
Please let us know in a comment if you can still reproduce the issue in 2.1.1, and we'll reopen it

**Comment by CombatCode** [comment]
Same problem, using:
Rancher Server 2.1.1
Rancher CLI 2.0.4, 2.0.5, 2.0.6.rc2

**Comment by loganhz** [comment]
How did you create the cluster? Node template?

**Comment by ajfriesen** [comment]
Node template for aws ec2 instances.

**Comment by CombatCode** [comment]
Same issue with rancher/rancher: 2.1.2
hosts running on EC2

**Comment by tfiduccia** [comment]
@deniseschannon - this is indeed an issue with RKE amazon EC2 clusters.  I cannot ssh into the nodes using the ID or the node Name.

I tried with a DO cluster and it worked just fine.

**Comment by dramich** [comment]
Available in v2.3.3-rc1

**Comment by aaronRancher** [comment]
The bug was reproduced with Rancher server `rancher\rancher:v2.1.2`
CLI version `v2.0.6`

Steps:
Provision a cluster using EC2.
Connect to project using CLI.
SSH into EC2 node.

Run `aaronyee$ ./rancher ssh aaron-ecw1`
```
ssh: connect to host 172.31.9.173 port 22: Network is unreachable
```
SSH cannot connect using private IP.

The bug fix is verified with Rancher server `rancher\rancher:master-head` `version 6717e73d1`
CLI version `v2.4.0-rc1`

`rancher ssh -e <Node name>`

`v2.4.0-rc1` introduced the -e command which uses the external IP address to SSH into a node.
